### PR TITLE
chore(components/ag-grid): data grid strict mode support

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

**Comment by Blackbaud-PaulCrowder** [comment]
[AB#2195397](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195397)

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/821/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/821/storybooks/ag-grid-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/821/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/821/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/821/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/821/playground/)

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-currency/cell-editor-currency.component.ts [hunk 1]
```diff
@@ -52,8 +55,7 @@ export class SkyAgGridCellEditorCurrencyComponent
     this.columnHeader = this.params.colDef.headerName;
     this.rowNumber = this.params.rowIndex + 1;
     this.columnWidth = this.params.column.getActualWidth();
-    this.rowHeightWithoutBorders =
-      this.params.node && this.params.node.rowHeight - 4;
+    this.rowHeightWithoutBorders = (this.params.node?.rowHeight as number) - 4;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-currency/cell-editor-currency.component.ts]: Isn't using `as number` the same as `this.params.node.rowHeight!`? What happens if `rowHeight` is undefined?

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-currency/cell-editor-currency.component.ts [hunk 2]
```diff
@@ -64,33 +66,39 @@ export class SkyAgGridCellEditorCurrencyComponent
...
-        break;
+
+    if (control) {
+      switch (this.#triggerType) {
+        case SkyAgGridCellEditorInitialAction.Delete:
+          control.setValue(undefined);
+          break;
+        case SkyAgGridCellEditorInitialAction.Replace:
+          control.setValue(
+            parseFloat(this.params?.charPress as string) || undefined
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-currency/cell-editor-currency.component.ts]: Same here, `as string`.

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-currency/cell-editor-currency.component.ts [hunk 3]
```diff
@@ -64,33 +66,39 @@ export class SkyAgGridCellEditorCurrencyComponent
...
+          );
+          break;
+        case SkyAgGridCellEditorInitialAction.Highlighted:
+        case SkyAgGridCellEditorInitialAction.Untouched:
+        default:
+          control.setValue(parseFloat(this.params?.value as string));
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-currency/cell-editor-currency.component.ts]: `as string`

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-datepicker/cell-editor-datepicker.component.ts [hunk 4]
```diff
@@ -34,68 +34,75 @@ export class SkyAgGridCellEditorDatepickerComponent
...
-        this.rowHeightWithoutBorders =
-          this.params.node && this.params.node.rowHeight;
+        this.rowHeightWithoutBorders = this.#params?.node?.rowHeight;
       } else {
-        this.columnWidthWithoutBorders = this.columnWidth - 2;
+        this.columnWidthWithoutBorders =
+          this.columnWidth === undefined ? undefined : this.columnWidth - 2;
         this.rowHeightWithoutBorders =
-          this.params.node && this.params.node.rowHeight - 3;
+          (this.#params?.node.rowHeight as number) - 3;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-datepicker/cell-editor-datepicker.component.ts]: `as number`

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-datepicker/cell-editor-datepicker.component.ts [hunk 5]
```diff
@@ -105,7 +112,7 @@ export class SkyAgGridCellEditorDatepickerComponent
    */
   public afterGuiAttached(): void {
     const datepickerInputEl = this.datepickerInput
-      .nativeElement as HTMLInputElement;
+      ?.nativeElement as HTMLInputElement;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-datepicker/cell-editor-datepicker.component.ts]: `as HTMLInputElement`

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-datepicker/cell-editor-datepicker.component.ts [hunk 6]
```diff
@@ -117,12 +124,12 @@ export class SkyAgGridCellEditorDatepickerComponent
           datepickerInputEl.dispatchEvent(new Event('change'));
         });
       datepickerInputEl.select();
-      datepickerInputEl.setRangeText(this.params.charPress);
+
+      const charPress = this.#params?.charPress as string;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-datepicker/cell-editor-datepicker.component.ts]: `as string`

File: libs/components/ag-grid/src/lib/modules/ag-grid/fixtures/ag-grid-row-delete.component.fixture.ts [hunk 7]
```diff
@@ -112,13 +114,18 @@ export class SkyAgGridRowDeleteFixtureComponent implements OnInit {
   }
 
   public filterName(): Promise<void> {
-    const filterChangedPromise = fromEvent(this.gridApi, 'filterChanged')
+    const filterChangedPromise = fromEventPattern(
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/fixtures/ag-grid-row-delete.component.fixture.ts:117]: I didn't know about `fromEventPattern`. Cool!

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**